### ISSUE TITLE
Add allow nil for validations

### ISSUE THREAD

**Issue by bdtomlin** [luckyframework/avram#293]
It would be nice to be able to do something like:

```crystal
validate_size_of zip_code, is: 5, allow_nil: true
validate_size_of title, min: 3, allow_nil: true
validate_inclusion_of age, in: [30, 40, 50], allow_nil: true
```

This would help in the case where you want to validate optional fields.

Fyi, I'd be glad to handle this in the near future. I'm busy finishing a couple of projects right now but hope to get back to contributing soon.

**Comment by paulcsmith** [comment]
I love the idea!

**Comment by wout** [comment]
Is anyone already working on this? I actually need it. If not, I will have a go.

**Comment by paulcsmith** [comment]
I’d say go for it! Thanks @wout

**Comment by paulcsmith** [comment]
Closed by #299

**Comment by bdtomlin** [comment]
@wout, sorry for my lack of response. Somehow I missed the conversation here. Thanks for handling this.
